### PR TITLE
Using fetcher pattern for Oauth2 access token.

### PR DESCRIPTION
**Pull request by joshirohit100**

### CODE REVIEW

File: packages/next-drupal/src/client.ts [hunk 1]
```diff
@@ -1359,14 +1359,23 @@ export class DrupalClient {
...
       headers: {
         Authorization: `Basic ${basic}`,
         "Content-Type": "application/x-www-form-urlencoded",
       },
       body: `grant_type=client_credentials`,
-    })
+    };
+
+    // Initializing empty response object.
+    let response = new Response();
```
Comment (lauriii) [inline packages/next-drupal/src/client.ts:1372]: I think this all could be done on a single line using ternary operator. Something like:

 ```
const response = this.fetcher ? await this.fetcher(url.toString(), options) : await fetch(url.toString(), options);
```

File: packages/next-drupal/src/client.ts [hunk 2]
```diff
@@ -1359,14 +1359,23 @@ export class DrupalClient {
 
     const basic = Buffer.from(`${clientId}:${clientSecret}`).toString("base64")
 
-    const response = await fetch(url.toString(), {
+    const options = {
```
Comment (lauriii) [inline packages/next-drupal/src/client.ts:1362]: For consistency, this is called `init` in the `this.fetch` because options refers to something else.

**Review by lauriii — CHANGES_REQUESTED** [review]

File: packages/next-drupal/src/client.ts (hunk 2 above)
Comment (joshirohit100) [reply packages/next-drupal/src/client.ts]: This is updated!

File: packages/next-drupal/src/client.ts (hunk 1 above)
Comment (joshirohit100) [reply packages/next-drupal/src/client.ts:1372]: Yes! we can use the ternary but I used the pattern same as https://github.com/chapter-three/next-drupal/blob/main/packages/next-drupal/src/client.ts#L306 with the debug message

**Comment by shadcn** [comment]
I think replacing this line is all we need to do here: https://github.com/chapter-three/next-drupal/blob/99bf2b6afaf55d5955d03971d186b7da42ce8f35/packages/next-drupal/src/client.ts#L1362

```diff
- const response = await fetch(url.toString(), {
+ const response = await this.fetch(url.toString(), { 
```

`client.fetch` already takes care of calling the default or the custom fetcher.

**Comment by joshirohit100** [comment]
@shadcn calling this.fetch() not causing recursion as then it will again perform all those checks and call getAccessToken()?

**Comment by joshirohit100** [comment]
PR is updated to use `this.fetch`

**Review by shadcn — APPROVED** [review]

**Pull request merged**